### DISCUSSION TITLE
Get the raw, static SQL query while writing code (and not at runtime)

### DISCUSSION THREAD

**Discussion by bennettdams** [prisma/prisma#20564] [Q&A] [answered]
### Question

Hey!
I have an idea for an educational tool which I need the raw SQL query for.

Is it possible to see the raw SQL query Prisma is going to execute?
So when I have a query like so:
```ts
await ctx.prisma.post.findMany({ select: { title: true }, take: 10 })
```

...I would like to also get the information how the raw SQL query looks like, here:
```sql
SELECT "public"."Post"."id", "public"."Post"."title"
FROM "public"."Post"
WHERE 1=1
ORDER BY "public"."Post"."id" ASC
LIMIT $1 OFFSET $2
```

I know I can get this information from the [log level config](https://www.prisma.io/docs/concepts/components/prisma-client/working-with-prismaclient/logging) at runtime, but is it also possible to get this while writing code?
I looked at the new Prisma Client extensions, but without luck.

**Comment by janpio** [comment, marked as answer]
Unfortunately not yet: https://github.com/prisma/prisma/issues/5052
